src/Platform/PlatformStorage.php

```diff
@@ -0,0 +1,200 @@
+    $this->filesystem = $filesystem;
+    $this->dispatcher = $dispatcher;
+    $this->runner = $runner;
+    $this->container = $container;
+  }
+
+  /**
+   * Loads a platform object by alias.
+   *
+   * @param string $alias
+   *   The alias of the platform to load.
+   *
+   * @return \EclipseGc\CommonConsole\PlatformInterface|null
+   *   The platform object on success. NULL if it doesn't exist.
+   */
+  public function load(string $alias) : ?PlatformInterface {
+    $directory = $this->ensureDirectory();
+    $alias_file = implode(DIRECTORY_SEPARATOR, [$directory, "$alias.yml"]);
+    if (!$this->filesystem->exists($alias_file)) {
+      return NULL;
+    }
+    $config = new Config(Yaml::parse(file_get_contents($alias_file)));
+    $platform_event = new GetPlatformTypeEvent($config->get(PlatformInterface::PLATFORM_TYPE_ID));
+    $this->dispatcher->dispatch(CommonConsoleEvents::GET_PLATFORM_TYPE, $platform_event);
+    return $this->getPlatform($platform_event, $config);
+  }
+
+  /**
+   * Checks to see if a specific platform exists.
+   *
+   * @param string $name
+   *   The platform to check to see if it exists.
+   *
+   * @return bool
+   */
+  public function exists(string $name) : bool {
+    $directory = $this->ensureDirectory();
+    $alias_file = implode(DIRECTORY_SEPARATOR, [$directory, "$name.yml"]);
+    return $this->filesystem->exists($alias_file);
+  }
+
+  /**
+   * Save a platform object.
+   *
+   * @param \EclipseGc\CommonConsole\PlatformInterface $platform
+   *   The platform to save.
+   *
+   * @return \EclipseGc\CommonConsole\PlatformInterface|null
+   *   Returns a fully loaded platform object on success. Null on failure.
+   */
+  public function save(PlatformInterface $platform) : ?PlatformInterface {
+    $directory = $this->ensureDirectory();
+    $alias = $this->getAlias($platform);
+    $alias_file = implode(DIRECTORY_SEPARATOR, [$directory, "{$alias}.yml"]);
+    try {
+      $this->filesystem->dumpFile($alias_file, Yaml::dump($platform->export()));
+      return $this->load($alias);
+    }
+    catch (IOException $e) {}
+  }
+
+  /**
+   * Delete a platform.
+   *
+   * @param \EclipseGc\CommonConsole\PlatformInterface $platform
+   *   The platform to delete.
+   *
+   * @return bool|null
+   */
+  public function delete(PlatformInterface $platform) : ?bool {
+    $directory = $this->ensureDirectory();
+    $alias_file = implode(DIRECTORY_SEPARATOR, [$directory, "{$this->getAlias($platform)}.yml"]);
+    if (!$this->filesystem->exists($alias_file)) {
+      return NULL;
+    }
+    try {
+      $this->filesystem->remove($alias_file);
+      return TRUE;
+    }
+    catch (IOException $e) {}
+    return FALSE;
+  }
+
+  /**
+   * Extracts the alias string from a platform.
+   *
+   * @param \EclipseGc\CommonConsole\PlatformInterface $platform
+   *   The platform from which to extract the alias.
+   *
+   * @return string
+   */
+  protected function getAlias(PlatformInterface $platform) : string {
```

Comment:
Would it make sense to move this to platform interface instead?

Reply:
And at the same time implement it in PlatformBase.

Reply:
Done